### PR TITLE
Add unit test for P2P base peer file - Closes #4223

### LINKED ISSUE #4223: Add unit test coverage for P2P base peer file
### Expected behavior

Every file from `lisk-p2p` library is well covered with unit test. The minimum must be 80% at least to achieve green reports.  

### Actual behavior

`peer/base.ts` counts with only  77.55 %.

### Steps to reproduce

`npm t`

### Which version(s) does this affect? (Environment, OS, etc...)

`+2.3`

### PR DESCRIPTION
**Pull request by diego-G**
### What was the problem?

Coverage was under 80%.

### How did I solve it?

Extending coverage beyond 90%. Every public method has been covered. Private ones were partially covered indirectly. Constructor is not well covered though.

### How to manually test it?

`npm t`

### Review checklist

- [x] The PR resolves #4223
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: elements/lisk-p2p/test/unit/peer/base.ts [hunk 1]
```diff
@@ -26,45 +51,627 @@ describe('peer', () => {
...
+				sandbox.stub(defaultPeer, 'request').resolves({
+					data: {
+						peers,
+						success: true,
+					},
+				});
+			});
+
+			it('should return a sanitized peer list', async () => {
+				const response = ((await defaultPeer.fetchPeers()) as unknown) as P2PResponsePacket;
```
Comment (ishantiw) [inline elements/lisk-p2p/test/unit/peer/base.ts]: why we are casting it to `P2PResponsePacket`? `defaultPeer.fetchPeers()` returns `ReadonlyArray<P2PPeerInfo>` so we don't have to typecast here

File: elements/lisk-p2p/test/unit/peer/base.ts [hunk 2]
```diff
@@ -26,45 +51,627 @@ describe('peer', () => {
...
+			});
+		});
+	});
+
+	describe('#request', () => {
+		it('should throw error if socket does not exists', async () => {
+			const p2pPacket = {
+				data: 'myData',
+				procedure: 'myProcedure',
+			} as P2PRequestPacket;
```
Comment (ishantiw) [inline elements/lisk-p2p/test/unit/peer/base.ts]: I think we don't need `as P2PRequestPacket` and in other places in this file

File: elements/lisk-p2p/test/unit/peer/base.ts [hunk 3]
```diff
@@ -14,57 +14,658 @@
...
+				const reputation = defaultPeer.reputation;
+				const penalty = DEFAULT_REPUTATION_SCORE / 10;
+				defaultPeer.applyPenalty(penalty);
+				expect(defaultPeer.reputation).to.be.eql(reputation - penalty);
+			});
+
+			it('should not ban peer', () => {
+				const penalty = DEFAULT_REPUTATION_SCORE / 10;
+				defaultPeer.applyPenalty(penalty);
+				expect(defaultPeer['_banPeer']).to.be.not.called;
```
[the remark this answers is not in the document]
Comment (diego-G) [reply elements/lisk-p2p/test/unit/peer/base.ts]: This is not mandatory. The IDE doesn't complain nor suggest anything.

File: elements/lisk-p2p/test/unit/peer/base.ts [hunk 4]
```diff
@@ -14,57 +14,658 @@
...
+
+			expect(defaultPeer.productivity)
+				.to.be.an('object')
+				.and.be.deep.equal(productivity);
+		}));
+
+	describe('#wsMessageRate', () =>
+		it('should get wsMessageRate property', () =>
+			expect(defaultPeer.wsMessageRate)
+				.to.be.a('number')
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/base.ts]: this is redundant and same for all similar cases

File: elements/lisk-p2p/test/unit/peer/base.ts [hunk 5]
```diff
@@ -14,57 +14,658 @@
...
+		it('should get productivity property', () => {
+			const productivity = {
+				requestCounter: 0,
+				responseCounter: 0,
+				responseRate: 0,
+				lastResponded: 0,
+			};
+
+			expect(defaultPeer.productivity)
+				.to.be.an('object')
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/base.ts]: this is redundant

File: elements/lisk-p2p/test/unit/peer/base.ts [hunk 6]
```diff
@@ -14,57 +14,658 @@
...
+				.and.be.deep.equal(productivity);
```
Comment (shuse2) [inline elements/lisk-p2p/test/unit/peer/base.ts]: I think `eql` is basically `deep.equal`

**Pull request merged**